### PR TITLE
IBP-3172 IBP-2986 IBP-2975 IBP-2985 IBP-3215 IBP-3133 - Inventory Manager

### PR DESCRIPTION
**Pull request by nahuel-soldevilla**

### CODE REVIEW

File: src/main/java/org/generationcp/middleware/dao/ims/LotDAO.java [hunk 1]
```diff
@@ -634,4 +639,96 @@ private void createLotRows(List<Lot> lots, Query query, boolean withReservationM
...
+			query.addScalar("createdByUsername");
+			query.addScalar("createdDate", Hibernate.DATE);
+			query.addScalar("lastDepositDate", Hibernate.DATE);
+			query.addScalar("lastWithdrawalDate",Hibernate.DATE);
+
+			query.setResultTransformer(Transformers.aliasToBean(LotDto.class));
+
+			if (pageable!= null) {
+				query.setFirstResult(pageable.getPageSize() * pageable.getPageNumber());
+				query.setMaxResults(pageable.getPageSize());
```
Comment (nahuel-soldevilla) [inline src/main/java/org/generationcp/middleware/dao/ims/LotDAO.java]: To add **sorting**, there is this method org.generationcp.middleware.dao.SampleListDao#addOrder that I've been thinking we can move below org.generationcp.middleware.dao.GenericDAO#addOrder to work as generic _pageable-to-sql_ **orderBy** utility

cc @clarysabel

File: src/main/java/org/generationcp/middleware/dao/ims/LotDAO.java [hunk 2]
```diff
@@ -10,32 +10,37 @@
...
-import java.util.Map;
-import java.util.Set;
-
 import com.google.common.collect.Lists;
-
 import com.google.common.collect.Sets;
 import org.generationcp.middleware.dao.GenericDAO;
 import org.generationcp.middleware.domain.inventory.LotAggregateData;
+import org.generationcp.middleware.domain.inventory_new.LotDto;
+import org.generationcp.middleware.domain.inventory_new.LotsSearchDto;
```
Comment (nahuel-soldevilla) [inline src/main/java/org/generationcp/middleware/dao/ims/LotDAO.java]: Maybe `org.generationcp.middleware.domain.inventorymanager` or `org.generationcp.middleware.domain.inventory.manager` or just everything inside `org.generationcp.middleware.domain.inventory`

File: src/main/java/org/generationcp/middleware/dao/ims/TransactionDAO.java [hunk 3]
```diff
@@ -545,4 +557,217 @@ private void createTransactionRow(final List<TransactionReportRow> transactionRe
 		}
 
 	}
+
+	//New inventory functions, please locate them below this line to help cleaning in the near future.
+	private final String SEARCH_TRANSACTIONS_QUERY = "SELECT " //
+		+ "    act.trnid AS transactionId,"//
+		+ "    users.uname AS user,"//
+		+ "    (CASE"//
+		+ "        WHEN trnstat = 1 AND trnqty >= 0 THEN 'Deposit'"//
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/dao/ims/TransactionDAO.java]: @vmaletta we should use TransactionStatus and TransactionType enum in the new queries. Please fix SEARCH_TRANSACTIONS_QUERY and appends

File: src/main/java/org/generationcp/middleware/dao/ims/LotDAO.java [hunk 4]
```diff
@@ -633,6 +645,249 @@ private void createLotRows(List<Lot> lots, Query query, boolean withReservationM
 		}
 	}
 
+	//New inventory functions, please locate them below this line to help cleaning in the near future.
+	private final String SEARCH_LOT_QUERY = "SELECT lot.lotid as lotId, " //
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/dao/ims/LotDAO.java]: @clarysabel 
Use enums

File: src/main/java/org/generationcp/middleware/dao/ims/TransactionDAO.java (hunk 3 above)
Comment (vmaletta) [reply src/main/java/org/generationcp/middleware/dao/ims/TransactionDAO.java]: Ok

File: src/main/java/org/generationcp/middleware/domain/inventory/manager/TransactionDto.java [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+import org.pojomatic.annotations.AutoProperty;
+
+import java.util.Date;
+
+@AutoProperty
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class TransactionDto {
+
+	private Integer transactionId;
+	private String user;
```
Comment (clarysabel) [inline src/main/java/org/generationcp/middleware/domain/inventory/manager/TransactionDto.java]: @vmaletta @cuenyad rename to createdByUsername

File: src/main/java/org/generationcp/middleware/dao/ims/LotDAO.java (hunk 1 above)
Comment (clarysabel) [reply src/main/java/org/generationcp/middleware/dao/ims/LotDAO.java]: 👍

**Pull request merged**